src/app/content/utils/processBrowserRedirect.ts

```diff
@@ -2,21 +2,28 @@ import { History } from 'history';
 import { Redirects } from '../../../../data/redirects/types';
 import { RouterService } from '../../navigation/routerService';
 import { assertWindow } from '../../utils';
-import { matchForRoute } from '../../navigation/utils';
-import { external } from '../../errors/routes';
+import { Dispatch } from '../../types';
+import { replace } from '../../navigation/actions';
+import { AnyMatch } from '../../navigation/types';
 
-export const processBrowserRedirect = async(services: {router: RouterService, history: History}) => {
+export const processBrowserRedirect = async(services: {
+  router: RouterService,
+  history: History,
+  dispatch: Dispatch
+}) => {
   const window = assertWindow();
+  const base = window.location.origin;
+
   const redirects: Redirects = await fetch('/rex/redirects.json')
     .then((res) => res.json())
     .catch(() => []);
 
   for (const {from, to} of redirects) {
     if (from === services.history.location.pathname) {
-      if (matchForRoute(external, services.router.findRoute(to))) {
-        window.location.href = window.location.origin + to;
-      }
-      services.history.replace(to);
+      const {pathname, search, hash} = new URL(to, base);
```

Comment:
i would add the url parsing within `findRouteMatch` if the given location is a string. the option for it to be a string is something we added in your previous redirects change so that should be safe